public/docs/ts/latest/guide/router.jade

```diff
@@ -868,23 +899,192 @@ a#nav-to-list
+
+  The Component Router supports navigation with optional parameters as well as required route parameters.
+  We define _optional_ parameters in an *object* after we define our required route parameters.
+
+  ### Route Parameters: Required or Optional?
+
+  There is no hard-and-fast rule. In general,
+
+  *prefer a required route parameter when*
+  * the value is required.
+  * the value is necessary to distinguish one route path from another.
+
+  *prefer an optional parameter when*
+  * the value is complex and/or multi-variate.
+
+  <a id="route-parameters-object"></a>
+  ### Route parameter
+
+  When navigating to the `HeroDetailComponent` we specified the `id` of the hero-to-edit in the
+  *route parameter* and made it the second item of the [*link parameters array*](#link-parameters-array).
+
++makeExcerpt('app/heroes/hero-list.component.1.ts', 'link-parameters-array', '')
+
+:marked
+  The router embedded the `id` value in the navigation URL because we had defined it
+  as a route parameter with an `:id` placeholder token in the route `path`:
+
++makeExcerpt('app/heroes/heroes.routing.ts', 'hero-detail-route', '')
+
+:marked
+  When the user clicks the back button, the `HeroDetailComponent` constructs another *link parameters array*
+  which it uses to navigate back to the `HeroListComponent`.
+
++makeExcerpt('app/heroes/hero-detail.component.1.ts', 'gotoHeroes', '')
+
+:marked
+  This array lacks a route parameter because we had no reason to send information to the `HeroListComponent`.
+
+  Now we have a reason. We'd like to send the id of the current hero with the navigation request so that the
+  `HeroListComponent` can highlight that hero in its list.
+
+  We do that with an object that contains our optional `id` parameter.
+  We also defined a junk parameter (`foo`) that the `HeroListComponent` should ignore.
+  Here's the revised navigation statement:
+
++makeExcerpt('app/heroes/hero-detail.component.ts', 'gotoHeroes-navigate', '')
+
+:marked
+  The application still works. Clicking "back" returns to the hero list view.
+
+  Look at the browser address bar.
+.l-sub-section
+  img(src='/resources/images/devguide/plunker-separate-window-button.png' alt="pop out the window" align="right" style="margin-right:-20px")
+  :marked
+    When running in plunker, pop out the preview window by clicking the blue 'X' button in the upper right corner.
+:marked
+ It should look something like this, depending on where you run it:
+
+code-example(language="bash").
+  localhost:3000/heroes;id=15;foo=foo
+
+:marked
+  The `id` value appears in the URL as (`;id=15;foo=foo`), not in the URL path.
+  The path for the "Heroes" route doesn't have an `:id` token.
+
+  The optional route parameters are not separated by "?" and "&".
+  They are **separated by semicolons (;)**
+  This is *matrix URL* notation &mdash; something we may not have seen before.
+
+.l-sub-section
+  :marked
+    *Matrix URL* notation is an idea first floated
```

Comment:
I like this note/callout. Good job here. Should we also address any concern that since it is not HTML standard that it will still work in all browsers that A2 supports as A2 handles this?